### PR TITLE
fix Pandora errors by using reference together with auto

### PR DESCRIPTION
**Pull request by wenxingfang**
#208 this problem is fixed! The reason for the errors is that the reference (&) is missing for some cases when using auto.

### CODE REVIEW

**Comment by wenxingfang** [comment]
@mirguest Could you help to review this PR? Thanks!

File: Reconstruction/PFA/Pandora/GaudiPandora/src/CaloHitCreator.cpp [hunk 1]
```diff
@@ -431,8 +430,7 @@ pandora::StatusCode CaloHitCreator::CreateHCalCaloHits(const CollectionMaps& col
             {
                 try
                 {
-                    auto pCaloHit0 = pCaloHitCollection.at(i);
-                    auto pCaloHit = &(pCaloHit0);
+                    auto pCaloHit = const_cast<edm4hep::CalorimeterHit*>(&(pCaloHitCollection.at(i)));
```
Comment (mirguest) [inline Reconstruction/PFA/Pandora/GaudiPandora/src/CaloHitCreator.cpp:433]: Why need to use the `const_cast`?

Comment (wenxingfang) [reply Reconstruction/PFA/Pandora/GaudiPandora/src/CaloHitCreator.cpp]: You are right, now the const_cast is removed.

**Review by mirguest — APPROVED** [review]
Thanks.

**Pull request merged**